tests/test_salesforce_sync_canary.py

```diff
@@ -44,10 +45,19 @@ def test_run(self):
         # run in check mode
         found_catalogs = self.run_and_verify_check_mode(conn_id)
 
+        # partition the found catalogs into 7 groups, 1 for each day of the week to save on both
+        #  time and API quota
+        weekday = datetime.weekday(datetime.now())  # weekdays 0-6, Mon-Sun
+        partition_size = math.ceil(len(found_catalogs)/7)
+        # buffer each side of the slice to account for dynamic stream discovery
+        start_of_slice = max(partition_size * weekday - 10, 0)
+        end_of_slice = min(partition_size * (weekday + 1) + 10, len(found_catalogs) + 1)
```

Comment:
```suggestion
        end_of_slice = min(partition_size * (weekday + 1) + 10, len(found_catalogs))
```

Correct me if I'm wrong, but the `+ 1` is pointless.
```python
>>> found_catalogs = [0, 1, 2, 3, 4]
>>> len(found_catalogs)
5
>>> found_catalogs[:5]
[0, 1, 2, 3, 4]
>>> found_catalogs[:6]
[0, 1, 2, 3, 4]
```

Reply:
Correct, I was thinking index rather than length.  Will remove the +1